docs/extensions/guides/ipc.md

```diff
@@ -0,0 +1,92 @@
+# Inter Process Communication
+
+A Lens Extension can utilise IPC to send information between its `LensRendererExtension` and its `LensMainExtension`.
+This is useful when wanting to communicate directly within your extension.
+For example, if a user logs into a service that your extension is a facade for and `main` needs to know some information so that you can start syncing items to the `Catalog`, this would be a good way to send that information along.
+
+IPC channels are blocked off per extension.
+Meaning that each extension can only communicate with itself.
+
+## Types of IPC
+
+There are two flavours of IPC that are provided:
+
+- Event based
+- Request based
+
+### Event Based IPC
+
+This is the same as an [Event Emitter](https://nodejs.org/api/events.html#events_class_eventemitter) but is not limited to just one Javascript process.
+This is a good option when you need to report that something has happened but you don't need a response.
+
+This is a fully two-way form of communication.
+Both `LensMainExtension` and `LensRendererExtension` can do this sort of IPC.
+
+### Request Based IPC
+
+This is more like a Remote Procedure Call (RPC).
+With this sort of IPC the caller waits for the result from the other side.
+This is accomplished by returning a `Promise<T>` which needs to be `await`-ed.
+
+This is a unidirectional form of communication.
+Only `LensRendererExtension` can initiate this kind of request, and only `LensMainExtension` and respond this this kind of request.
+
+## Registering IPC Handlers and Listeners
+
+The general terminology is as follows:
+
+- A "handler" is the function that responds to a "Request Based IPC" event.
+- A "listener" is the function that is called when a "Event Based IPC" event is emitted.
+
+To register either a handler or a listener, you should do something like the following:
+
+```typescript
+import { LensMainExtension, Interface, Types } from "@k8slens/extensions";
+
+export class ExampleExtensionMain extends LensMainExtension {
+  onActivate() {
+    this.disposers.push(
+      this.listenIpc({
+        channel: "initialize",
+        listener: this.initializeListener,
+        verifier: this.initializeVerifier,
+      })
+    );
+  }
+
+  initializeListener = (event: Types.IpcMainEvent, uid: string) => {
+    console.log(`starting to initialize: ${uid}`);
+  };
+
+  initializeVerifier = (args: unknown[]): args is [uid: string] => {
+    return args.length === 1 && typeof args[0] === "string";
+  }
+}
+```
```

Comment:
Do you do this very same thing from `LensRendererExtension` and then from `main` call `this.sendIpc("channel", ...args)`? Does it matter if the channel names are different? Or listeners are a way for `main` to push to `renderer` as well as, optionally, `renderer` to push to `main`?

Reply:
The channel names have to be the same and you have to register a "listener" on `renderer` for events sent from `main` to be known about.

Reply:
I think that needs to be clarified in the docs, then. The docs are only telling one side of the story.